vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/NetworkActions.java

```diff
@@ -0,0 +1,184 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.mobileapp.action;
+
+import java.util.Optional;
+
+import org.openqa.selenium.WebElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.manager.GenericWebDriverManager;
+import org.vividus.steps.ui.validation.IBaseValidations;
+import org.vividus.ui.action.search.Locator;
+import org.vividus.ui.mobile.action.search.AppiumLocatorType;
+
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.android.connection.ConnectionStateBuilder;
+
+public class NetworkActions
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkActions.class);
+
+    private static final String IOS_PREFERENCES_BUNDLE_ID = "com.apple.Preferences";
+
+    private final IWebDriverProvider webDriverProvider;
+    private final GenericWebDriverManager genericWebDriverManager;
+    private final ApplicationActions applicationActions;
+    private final IBaseValidations baseValidations;
+
+    public NetworkActions(IWebDriverProvider webDriverProvider, GenericWebDriverManager genericWebDriverManager,
+            ApplicationActions applicationActions, IBaseValidations baseValidations)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.genericWebDriverManager = genericWebDriverManager;
+        this.applicationActions = applicationActions;
+        this.baseValidations = baseValidations;
+    }
+
+    public void changeNetworkConnectionState(NetworkToggle networkToggle, Mode mode)
+    {
+        if (genericWebDriverManager.isAndroid())
+        {
+            webDriverProvider.getUnwrapped(AndroidDriver.class)
+                    .setConnection(NetworkToggle.ON == networkToggle ? mode.getEnablingConnectionStateBuilder().build()
+                            : mode.getDisablingConnectionStateBuilder().build());
+        }
+        if (genericWebDriverManager.isIOS())
+        {
+            changeNetworkConnectionStateForIOS(networkToggle, mode);
+        }
+    }
+
+    private void changeNetworkConnectionStateForIOS(NetworkToggle networkToggle, Mode mode)
+    {
+        applicationActions.activateApp(IOS_PREFERENCES_BUNDLE_ID);
+        findMenuItemInSettings(new Locator(AppiumLocatorType.ACCESSIBILITY_ID, mode.getIOSNetworkConnectionAlias()))
+                .ifPresent(WebElement::click);
```

Comment:
what should happen if there is no such item in iOS preferences? does it make sense to proceed with next actions in this method?